### ISSUE TITLE
bad location for "must not have a part-of directive" error message

### ISSUE THREAD

**Issue by skybrian** [dart-lang/sdk#25709]
To reproduce:

entry.dart
```
library entry;

part 'part1.dart';
part 'part2.dart';
```

part1.dart
```
// line 1 comment
part of entry; // line 2 comment
// line 3 comment
// line 4 comment
```

part2.dart
```
// line 1 comment
// line 2 comment
// line 3 comment
import "part1.dart";
```

One of the errors that appears is:
The imported library '"part1.dart"' must not have a part-of directive - bin/badparts/part1.dart, line 4

However, the line number is incorrect, since part1.dart only has a comment on line 4.

In Atom, the range underlined in red in part1.dart corresponds to the location of the import statement in part2.dart. (If you click on the error, Atom highlights a range starting in the middle of line 3 and ending in the middle of line 4.)

I suspect that when we create this error message, we are combining the filename from one file involved in the error with a character offset from the other file involved in the error.

**Comment by bwilkerson** [comment]
Yes. This error should be associated with the file 'part2.dart' (the file containing the import directive). It should probably be re-worded to indicate that it's the import directive that's in error, not the part-of directive.

**Comment by keertip** [comment]
This issue is obsolete. In the modern analyzer implementation, the `import_of_non_library` diagnostic is correctly reported on the import directive inside the importing file, rather than incorrectly targeting the part file. Additionally, the problem and correction messages have been updated to clearly indicate that the import directive is invalid.

If there is more work to do here, please let us know by filing a new issue with up to date information. Thanks!
